src/components/PolicyTopologyPage.tsx

```diff
@@ -670,4 +687,97 @@ const PolicyTopologyPage: React.FC = () => {
   );
 };
 
+/**
+ * Retrieves a mapping from Kubernetes resource kinds to their Group/Version/Kind (GVK)
+ * by querying both core API resources and aggregated API discovery.
+ *
+ * TODO: externalise this as a more general purpose helper for use elsewhere in the plugin
+ * TODO: consider a contrib to https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-utils/src/k8s/k8s-utils.ts
+ *
+ * This function accepts a resource hints object. If multiple API groups provide the same kind,
+ * the one matching the hint (if provided) will be used.
+ *
+ * @param resourceHints - a mapping of resource kinds to the preferred API group.
+ * Example:
+ * {
+ *   Gateway: 'gateway.networking.k8s.io',
+ *   DNSRecord: 'kuadrant.io'
+ * }
+ *
+ * @returns a Promise that resolves to an object mapping resource kinds to their GVK.
+ */
+export const getGroupVersionKindForKind = async (
+  resourceHints: Record<string, string>,
+): Promise<Record<string, GVK>> => {
+  const mapping: Record<string, GVK> = {};
+
+  // Helper to provide resource hinting where there could be resource ambiguity (E.g. `dnsrecords.kuadrant.io` vs `dnsrecords.ingress.operator.openshift.io`)
+  const updateMapping = (kind: string, group: string, version: string) => {
+    if (resourceHints[kind]) {
+      // Always override if the new group is the hinted group.
+      if (group === resourceHints[kind]) {
+        mapping[kind] = { group, version, kind };
+      }
+    } else if (!mapping[kind]) {
+      mapping[kind] = { group, version, kind };
```

Comment:
late to the PR, but question about this line to confirm my understanding.
This will use the first found match of the kind, unless it's one of the hinted kinds?

Reply:
np: yes